engine/src/org/pentaho/di/trans/steps/http/HTTP.java

```diff
@@ -171,11 +175,21 @@ public HTTP( StepMeta stepMeta, StepDataInterface stepDataInterface, int copyNr,
                   encoding = contentType.replaceFirst( "^.*;\\s*charset\\s*=\\s*", "" ).replace( "\"", "" ).trim();
                 }
               }
+              
+              headerString.append("{");
```

Comment:
Building a JSON Object using a string builder is a bad idea. If one of the values contains a double-quote (or other characters needing escaped), this will fail. Please see the way JSONObject is used in the JsonOutput step.